src/odemis/driver/actuator.py

```diff
@@ -2179,3 +2185,631 @@ def terminate(self):
+        # listen to reference changes from the underlying stage
+        self._lensz.referenced.subscribe(self._onLensReferenced, init=True)
+
+    def _onLensReferenced(self, cref):
+        """
+        Update the referenced value from the underlying lens stage
+        """
+        # Directly set the value from the updated argument
+        if 'z' in cref:
+            self.referenced._set_value({'z': cref['z']}, force_write=True)
+
+    def updateMetadata(self, md):
+        # Prevent manual update of focus FAV_POS_ACTIVE and FAV_POS_DEACTIVE
+        if model.MD_FAV_POS_ACTIVE in md or model.MD_FAV_POS_DEACTIVE in md:
+            raise ValueError("Focus FAV_POS_ACTIVE and FAV_POS_DEACTIVE cannot be set manually.")
+
+        super(LinkedHeightFocus, self).updateMetadata(md)
+        # Re-update focus position if POS_COR is modified
+        if model.MD_POS_COR in md:
+            self._updatePosition(self._lensz.position.value)
+
+    def _updatePosition(self, lens_pos):
+        """
+        Update the position VA from the underlying dependency
+        :param lens_pos: (dict string-> float) the lens Z value
+        """
+        # Only update position when MD_POS_COR is already configured
+        if model.MD_POS_COR not in self._metadata:
+            logging.error("Focus POS_COR is not found in metadata.")
+            return
+        if util.almost_equal(lens_pos['z'], self._lensz.getMetadata()[model.MD_FAV_POS_DEACTIVE]['z'], atol=1e-3):
+            # Set focus FAV_POS_DEACTIVE when the lens pos is FAV_POS_DEACTIVE
+            focus_val = self._metadata[model.MD_FAV_POS_DEACTIVE]['z']
+        else:
+            focus_val = self._getFocusValue(target_lensz=lens_pos['z'])
+        logging.debug("Updating focus position %s.", focus_val)
+        self.position._set_value({'z': focus_val}, force_write=True)
+
+    def _isInRange(self, pos=None):
+        """
+        A helper function to check if current position is in focus range
+        :return: True if position in focus range, False otherwise
+        """
+        pos = self.position.value['z'] if pos is None else pos
+        # Add 1% margin for hardware slight errors
+        margin = (self._range[1] - self._range[0]) * 0.01
+        return (self._range[0] - margin) <= pos <= (self._range[1] + margin)
+
+    @isasync
+    def moveAbs(self, pos):
+        """
+        Move the focus to the defined position for the Z axis. This is an
+        asynchronous method.
+        pos dict(string-> float): name of the axis and new position in m
+        returns (Future): object to control the move request
+        """
+        if not pos:
+            return model.InstantaneousFuture()
+        # check pos value is in range (except when it's = deactive)
+        if not self._isInRange(pos['z']) and not util.almost_equal(pos['z'],
+                                                                   self._metadata[model.MD_FAV_POS_DEACTIVE]['z'],
```

Comment:
You have several places checking for a position being (almost) equal to the deactive. I think it'd make it a little clearer if you had a dedicated function, something like `_isDeactivated()` (or `_isParked()` ?). Quite similar to `_isInRange()`.